frontend/src/pages/ApartmentPage.tsx

```diff
@@ -205,14 +213,15 @@ const ApartmentPage = (): ReactElement => {
   const removeLike = likeHelper(true);
 
   const openReviewModal = async () => {
+    let user = await getUser(true);
     if (!user) {
-      let user = await getUser(true);
       setUser(user);
       if (!user) {
         showSignInErrorToast();
         return;
       }
     }
+    setUser(user);
```

Comment:
You always `setUser(user)` no matter the if statement above result in, please move this line up before if statement, then delete the other getUser